### PR TITLE
Add CI (pre-commit.ci) for running pre-commit as CI

### PR DESCRIPTION
**Pull request by euronion**
the pre-commit.ci CI will automatically run pre-commit config via CI pre-commit-ing PRs.

TODO: Add correct link to pre-commit.ci badge. @fneum 

## Change proposed in this Pull Request

n/a

## Description

n/a

## Motivation and Context

Easier than GitHub Actions.

## How Has This Been Tested?

In different repository.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I tested my contribution locally and it seems to work fine.
- [ ] I locally ran `pytest` inside the repository and no unexpected problems came up.
- [ ] I have adjusted the docstrings in the code appropriately.
- [ ] I have documented the effects of my code changes in the documentation `doc/`.
- [ ] I have added newly introduced dependencies to `environment.yaml` file.
- [ ] I have added a note to release notes `doc/release_notes.rst`.
- [ ] I have used `pre-commit run --all` to lint/format/check my contribution

### CODE REVIEW

**Comment by FabianHofmann** [comment]
Due to reasonable concerns brought up in https://github.com/pre-commit/pre-commit-hooks/issues/291, we should not add pytest to the commit hooks (commits should be fast). So I'd say, we set up GH actions for testing and deploying (instead of travis) side by side with pre-commit ci.

File: README.rst [hunk 1]
```diff
@@ -169,4 +169,7 @@ See the individual files for license details.
    :target: https://api.reuse.software/info/github.com/pypsa/atlite
 .. |black| image:: https://img.shields.io/badge/code%20style-black-000000.svg
    :target: https://github.com/psf/black
-   :alt: Code style: black
\ No newline at end of file
+   :alt: Code style: black
+.. |pre-commit.ci| image:: https://results.pre-commit.ci/badge/github/euronion/atlite/master.svg
+   :target: https://results.pre-commit.ci/latest/github/euronion/atlite/master
+   :alt: pre-commit.ci status
```
Comment (fneum) [inline README.rst:175]:
```rst
.. |pre-commit.ci| image:: https://results.pre-commit.ci/badge/github/PyPSA/atlite/master.svg
   :target: https://results.pre-commit.ci/latest/github/PyPSA/atlite/master
   :alt: pre-commit.ci status
```

**Comment by fneum** [comment]
> TODO: Add correct link to pre-commit.ci badge. @fneum

see suggestion, as if already in `master`.

File: README.rst (hunk 1 above)
Comment (FabianHofmann) [reply README.rst:175]: already done in masterr, sorry, I was a bit fast here...

**Pull request merged**